wavecrestweb/kenworthy-machine
# #KM-17: Add footer

## Description

This PR creates and adds the footer to the `layout.tsx`. It also extends the custom theme to include:

- sizes and a new variant for Link component
~- sizes for Text component~
- a new variant for the Heading component

The spacing and font sizes are responsive for medium and large screens.
Currently, the content for the footer is hardcoded. Once, the CMS is set up, this component will be updated to fetch content from CMS. 

**UPDATE:** Refactored to address feedback:
- switched to relative units
- added responsiveness for logo
- added responsiveness for small screen sizes

## Related Tickets & Documents
Closes #KM-17

## Acceptance Criteria

- [x] The footer includes all necessary contact information.
- [x] The footer includes links to other pages are working and clearly labeled.
- [x] The footer navigation buttons have a red line appear below the word on hover.
- [x] The footer's design is consistent with the rest of the site.


## What type of PR is this? (check all applicable)


|     | Type                       |
| --- | -------------------------- |
|  `✓`   | :sparkles: New feature     |
|  `✓`   | 🎨 Style                   |
|     | :hammer: Refactoring       |
|     | 🔥 Performance Improvements|
|     | :bug: Bug fix              |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |


## Mobile & Desktop Screenshots/Recordings

![Screen Shot 2024-01-30 at 1 45 19 PM](https://github.com/wavecrestweb/kenworthy-machine/assets/109827834/f42a0b78-e3ac-4046-86c9-af5525f62474)
![Screen Shot 2024-01-30 at 1 45 38 PM](https://github.com/wavecrestweb/kenworthy-machine/assets/109827834/7240b394-f28a-43f5-8506-e2aa29e78072)
![Screen Shot 2024-01-30 at 2 56 43 PM](https://github.com/wavecrestweb/kenworthy-machine/assets/109827834/dbca28c8-5613-4e73-af62-4eb17dd8e726)

## What I learned
I dove into learning the ins-and-outs of ChakraUI and specifically customizing themes to match the design system. Learned about Next.js Image component!

## Testing steps

1. Scroll to the bottom of the page
2. Click on all the navigation links to check that redirection works as expected
3. Switch between different screen sizes to see how the spacing and font sizes change


## Added to documentation?

|     | Type                       |
| --- | -------------------------- |
|     | 📜 README.md    |
| `✓`    | 🙅 no documentation needed    |


## What gif best describes this PR?

  ![Finally](https://media.giphy.com/media/3oEjI53nBYOOEQgDcY/giphy.gif?cid=ecf05e47i9r3dojm2qz9p8vxlmfqeo37eutyivimwdpfwrx4&ep=v1_gifs_search&rid=giphy.gif&ct=g)
